provider/pkg/schema/gen.go

```diff
@@ -1347,3 +1369,35 @@ func GatherSemantics(schemaDir string) (SemanticsSpecDocument, error) {
 	}
 	return semanticsDocument, nil
 }
+
+type Reports struct {
```

Comment:
nit. I think reports should be in it's own file, gen.go is already a bit of a monster.